### PR TITLE
process_single changes

### PR DESCRIPTION
**Pull request by muffato**
`nf-core lint` complains that it doesn't know `process_single`:
1. Added `process_single` to the list of standard labels
2. `custom/dumpsoftwareversions/main.nf` is a `process_single` in nf-core/modules

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

**Review by jfy133 — CHANGES_REQUESTED** [review]
Missing CHANGELOG update

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
   - `Checks.groovy` -> `Utils.groovy`
   - `Completion.groovy` -> `NfcoreTemplate.groovy`
   - `Workflow.groovy` -> `WorkflowMain.groovy`
+- The `process_single` label is now allowed
```
Comment (jfy133) [inline CHANGELOG.md:19]:
```suggestion
- Modules using a `process_single` label will not throw a warning
```

**Review by jfy133 — COMMENTED** [review]
Tests need to pass first I gues?

**Comment by muffato** [comment]
I was under the impression the errors were not related to this change, but I'll take a look

**Comment by muffato** [comment]
OK. I found why the test fail.

Let's take `test_modules:test_mod_json_create` as an example. It calls `ModulesJson.create` there

https://github.com/nf-core/tools/blob/bc56cdd35077e408a0a7c90ec07fcdff7c2d1f3b/tests/modules/modules_json.py#L51

which calls this function on the names of the local modules
https://github.com/nf-core/tools/blob/bc56cdd35077e408a0a7c90ec07fcdff7c2d1f3b/nf_core/modules/modules_json.py#L76-L78

This function will check each of them against the remote to identify the branch / SHA.

https://github.com/nf-core/tools/blob/bc56cdd35077e408a0a7c90ec07fcdff7c2d1f3b/nf_core/modules/modules_json.py#L216
https://github.com/nf-core/tools/blob/bc56cdd35077e408a0a7c90ec07fcdff7c2d1f3b/nf_core/modules/modules_json.py#L227

Here,`custom/dumpsoftwareversions` doesn't match the `master` branch of the remote, since I've got a change in this pull-request, and it decides to prompt the user for the branch name, which fails under pytest because stdin is disabled

https://github.com/nf-core/tools/blob/bc56cdd35077e408a0a7c90ec07fcdff7c2d1f3b/nf_core/modules/modules_json.py#L235-L239

I don't know how to restructure the test so that it would just work.

To confirm that's indeed the issue, I made a local branch that skips the `custom/dumpsoftwareversions` change, and checked that it indeeds passes the test.

**Comment by SPPearce** [comment]
I think this was already incorporated into the code when the modules restructuring occurred.

**Comment by muffato** [comment]
You're right, someone else did this in parallel

**Pull request closed without merging**